add_node.rb

```diff
@@ -10,8 +11,37 @@
 fileData = File.read("data.json")
 obj = JSON.parse(fileData)
 
+# Initialise time periods.Dates have the format of yyyy-mm-dd
+this_year = Date.today.strftime("%Y")
+MID_SPRING_PERIOD = (this_year+"-02-15 to "+this_year+"-04-30").split(" to ")
+END_SPRING_PERIOD = (this_year+"-05-01 to "+this_year+"-09-30").split(" to ")
+MID_AUTUMN_PERIOD = (this_year+"-10-01 to "+this_year+"-11-30").split(" to ")
+END_AUTUMN_PERIOD = (this_year+"-12-01 to "+(this_year.to_i+1).to_s+"-02-14").split(" to ")
+
 puts "#{obj.length} objects earlier!"
 
+# Just make sure that batch is not "" as while loop will never execute
+batch = "h"
+batch_semester = ""
+batch_year = ""
+today = Date.today.to_s
+
+while !["y","Y","n","N"].include? batch
+	puts "Do you want to batch insert year & semester for the papers? [y/Y for Yes, n/N for No) "
+	batch = gets.chomp
+	if batch.downcase == "y"
+		puts "Enter batch semester : "
+		batch_semester = gets.chomp
```

Comment:
Here, `batch_semester` must match the regex for `mid-spring-2016`. This regex is probably there in one of the scripts inside mfqp-source, I will try to find it, you can run `grep -i mid` in the mfqp-source scripts folder and look through the results to find it. I would rather you reuse that regex because that has been checked by myself, and @xtinct as well, in an earlier PR.